thunder/distributed/tensor_parallel/column_wise.py

```diff
@@ -284,6 +284,7 @@ def forward(self, tokens: torch.Tensor) -> torch.Tensor:
             chunked_param_name_to_layer_type=chunked_param_name_to_layer_type,
             process_group=process_group,
         ),
+        _legacy_copy_params=True,
```

Comment:
I'd have tensor parallel transforms behave in the same way as fsdp v2 in regard to param handling; direct sharding w/o copies.

Reply:
also, if we go with this parameter being `True`, by default, which transforms want to copy params?

Reply:
Yes, this parameter is just a "let's keep Tensor Parallel without changing the transform in this PR", but we would want to update tensor parallel in the same way.

Reply:
I hope to get to the fourth part (delegate materialization to MaterializationTransform) ideally tomorrow, then we can see if we want to split the tensor parallel in the same parts or different ones.